src/api/app/services/github_status_reporter.rb

```diff
@@ -0,0 +1,52 @@
+class GithubStatusReporter < SCMStatusReporter
+  def call
+    github_client = Octokit::Client.new(access_token: @scm_token,
+                                        api_endpoint: @event_subscription_payload[:api_endpoint])
+    # https://docs.github.com/en/rest/reference/repos#create-a-commit-status
+    github_client.create_status(@event_subscription_payload[:target_repository_full_name],
+                                @event_subscription_payload[:commit_sha],
+                                @state,
+                                status_options)
+  rescue Octokit::InvalidRepository => e
+    package = Package.find_by_project_and_name(@event_payload[:project], @event_payload[:package])
+    return if package.blank?
+
+    tokens = Token::Workflow.where(scm_token: @scm_token).pluck(:id)
+    return if tokens.none?
+
+    EventSubscription.where(channel: 'scm', token: tokens, package: package).delete_all
+
+    @workflow_run.save_scm_report_failure("Failed to report back to GitHub: #{e.message}", request_context) if @workflow_run.present?
+  rescue Octokit::Error => e
+    rescue_with_handler(e) || raise(e)
+  rescue Faraday::ConnectionFailed => e
+    @workflow_run.update_as_failed("Failed to report back to GitHub: #{e.message}")
+  ensure
+    RabbitmqBus.send_to_bus('metrics', "scm_status_report,scm=#{@event_subscription_payload[:scm]},exception=#{e} value=1") if e.present?
+  end
+
+  private
+
+  # Depending on the SCM, the state is different
```

Comment:
This code is now specific to GitHub, so this comment (including the next line) should be removed. It doesn't provide value.